producer_client/lib/animals/resources/fake_cat.rb

```diff
@@ -3,16 +3,25 @@
 module Animals
   module Resources
     class FakeCat
+      @@cats = []
+
+      def self.reset!
+        @@cats = []
```

Comment:
With the introduction of resource creation, our fake must be upgraded with some in-memory storage.